### PR TITLE
HS-192: Add metrics api and push metrics from Minion RPC/heartbeat

### PR DESCRIPTION
**Pull request by cgorantla**
## Description
1. Add Metrics Adapter and push metrics from Minion RPC/Heartbeat
2. Refactor minion heartbeat from ipc -> minion

## Jira link(s)
- https://issues.opennms.org/browse/HS-192

## Checklist
* [x] Follows Horizon Stream's [development guidelines.](https://github.com/OpenNMS/horizon-stream/wiki/Development-Guidelines)
* [x] Appropriate reviewer(s) have been selected.
* [x] Jira issue(s) have been updated to "In Review".
* [ ] Includes [appropriate tests.](https://github.com/OpenNMS/horizon-stream/wiki/Test-Strategy)
* [ ] Documentation has been updated as necessary.

### CODE REVIEW

File: platform/ipc/echo/src/main/java/org/opennms/horizon/echo/monitor/SimpleMinionRpcMonitor.java [hunk 1]
```diff
@@ -142,9 +154,23 @@ public void process() {
...
-            log.info("ECHO RESPONSE: node-id={}; node-location={}; duration={}ms", nodeId, nodeLocation, responseTime);
+            LOG.info("ECHO RESPONSE: node-id={}; node-location={}; duration={}ms", nodeId, nodeLocation, responseTime);
+            updateMetrics(responseTime, nodeId, nodeId, nodeLocation);
         } catch (InterruptedException|ExecutionException t) {
-            log.warn("ECHO REQUEST failed", t);
+            LOG.warn("ECHO REQUEST failed", t);
+        }
+    }
+
+    private void updateMetrics(long responseTime, String minionId, String... labelValues) {
```
Comment (yangli2014) [inline platform/ipc/echo/src/main/java/org/opennms/horizon/echo/monitor/SimpleMinionRpcMonitor.java:164]: If it is not hard I suggest adding unit tests as well.

**Review by yangli2014 — COMMENTED** [review]
Excellent job and I tested it locally, it works as expected.

File: platform/ipc/echo/src/main/java/org/opennms/horizon/echo/monitor/SimpleMinionRpcMonitor.java (hunk 1 above)
Comment (cgorantla) [reply platform/ipc/echo/src/main/java/org/opennms/horizon/echo/monitor/SimpleMinionRpcMonitor.java:164]: Added for HeartbeatConsumer.

File: platform/metrics/impl/src/main/java/org/opennms/horizon/metrics/impl/OnmsMetricsAdapterImpl.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+package org.opennms.horizon.metrics.impl;
+
+import io.prometheus.client.Collector;
+import io.prometheus.client.exporter.PushGateway;
+import org.opennms.horizon.metrics.api.OnmsMetricsAdapter;
+
+import java.io.IOException;
+
+public class OnmsMetricsAdapterImpl implements OnmsMetricsAdapter {
```
[the remark this answers is not in the document]
Comment (cgorantla) [reply platform/metrics/impl/src/main/java/org/opennms/horizon/metrics/impl/OnmsMetricsAdapterImpl.java]: updated!

**Review by yangli2014 — APPROVED** [review]

**Pull request merged**